main/main.cpp

```diff
@@ -4108,8 +4108,20 @@ bool Main::iteration() {
 
 	RenderingServer::get_singleton()->sync(); //sync if still drawing from previous frames.
 
-	if (DisplayServer::get_singleton()->can_any_window_draw() &&
-			RenderingServer::get_singleton()->is_render_loop_enabled()) {
+	bool can_draw = DisplayServer::get_singleton()->can_any_window_draw();
+
+#ifndef _3D_DISABLED
+	if (!can_draw) {
+		Ref<XRInterface> xr_interface = XRServer::get_singleton()->get_primary_interface();
+		if (xr_interface.is_valid() && xr_interface->is_initialized()) {
```

Comment:
Is it possible for this condition to change at run time, or can this be cached?

My preference would be for the value to be cached in the DisplayServer and then the XRInterface  would essentially be treated as any other window and affect the result of `can_any_window_draw()`

Reply:
> Is it possible for this condition to change at run time, or can this be cached?

This can change from frame-to-frame, if the application enters or exits XR mode.

> My preference would be for the value to be cached in the DisplayServer and then the XRInterface would essentially be treated as any other window and affect the result of `can_any_window_draw()`

Hm. The DisplayServer is platform dependent, ie we have `DisplayServerWindows`, `DisplayServerX11`, `DisplayServerMacOS`, etc, whereas this check is platform independent.

I suppose we could put a function on the base class that combines the `XRServer` check with the platform specific check? But it still feels better to me to have the `XRServer` check outside of `DisplayServer`, to keep the platform dependent/independent split very clean.

Reply:
Great find David but I agree with Clay that this get very convoluted especially if we have other processes that want to overrule this. I can think of other use cases where we may want to render as per normal even if the application is minimized (think of our movie output, I might start that to generate a movie that last 15 minutes and minimize Godot while that is chugging away). 

I think we can turn this upside down and introduce a method on our OS class where OpenXR can register that there is another output that requires drawing. This needs to be made in such a way that multiple systems can register this simultaniously so it can't be a simple bool.

The code above would change to something like:
```
bool can_draw = DisplayServer::get_singleton()->can_any_window_draw() && OS::get_singleton()->can_any_output_draw();
```

And in OS we'd do something like:
```
bool OS::can_any_output_draw() {
  return !registered_output_draw_list.is_empty();
}

bool OS::register_output_draw(Ref<RefCounted> p_output) {
  if (!registered_output_draw_list.has(p_output)) {
    registered_output_draw_list.push_back(p_output);
  }
}

bool OS::unregister_output_draw(Ref<RefCounted> p_output) {
  if (registered_output_draw_list.has(p_output)) {
    registered_output_draw_list.erase(p_output);
  }
}
```

Or something along those lines

Then in `OpenXR::initialise` we call `register_output_draw` and in `OpenXR::uninitialize` we call `unregister_output_draw`